### ISSUE TITLE
Apply dark theme to maps.

### ISSUE THREAD

**Issue by patrickdalla** [sepinf-inc/IPED#1443]
Apply dark theme to maps.

**Comment by patrickdalla** [comment]
The implemented code was pushed to MapDarkTheme branch.

**Comment by patrickdalla** [comment]
It applies 3 filters to openstreet maps tile images: hue-rotate(180deg) brightness(80%) invert(1);
The initial hue-rotation keeps blue color (sea and rivers) still blue, and green color (some represented forests and parks) still green after the color inversion.

**Comment by patrickdalla** [comment]
For example:
The following original plotted map:
![image](https://user-images.githubusercontent.com/28692427/204884542-33a68714-65fb-47ab-a728-15045ef01c0c.png)
after the filters becomes:
![image](https://user-images.githubusercontent.com/28692427/204884666-8730112b-5e6a-4573-9088-462695d8e091.png)

**Comment by lfcnassif** [comment]
Great! Thank you @patrickdalla

**Comment by lfcnassif** [comment]
Just noticed the new map dark theme made map dragging/translation/zoom very very slow. I think it should be improved, so I'm reopening this. It is very slow in my machine, maybe it could be unusable in some user machine A simple but faster filter would be much better in my opinion.

**Comment by patrickdalla** [comment]
Yes, it consumes CPU as it applies image processing filter on every image (what I think makes the low battery usage supposed advantage of dark mode disappear). 
For now, I don't see other way. 
As the tile images are loaded from third parties, they should keep a 'dark' version, and IPED should change the tile server url to this new tile server. So IPED should keep some mapping between the light and dark url version of some tile server.
Maybe we can keep the map only in the user selected URL and not change it on IPED color theme change. So if the user wants a DARK version, he should configure a DARK tile server.

**Comment by lfcnassif** [comment]
I'm not sure if changing the tile server automatically is a good idea... If there is no faster filter to be applied, I prefer to disable the feature for now and use the faster light theme always. Is it ok for you?

**Comment by patrickdalla** [comment]
Yes. No problem.

Em ter., 24 de jan. de 2023 08:26, Luis Filipe Nassif <
***@***.***> escreveu:

**Comment by lfcnassif** [comment]
Just disabled it with commit above. Please double check it, since the merged PR mixed several features and I'm not sure which of them refer to dark theme. I recommend putting the issue number before each commit message, so we can find them easily to cherry-pick, revert, etc
